### ISSUE TITLE
Add Scikit-Learn

### ISSUE THREAD

**Issue by gcalmettes** [holzschu/Carnets#19]
Once Scipy is included, would Scikit-Learn be trivial to add? (dependencies Numpy/Scipy)

**Comment by holzschu** [comment]
I've had a look at it. scikit-learn is a really huge package.
It will require pandas and scipy, which are already a bit hard. Once scipy is done, I will try scikit-learn, but I can't guarantee anything.

**Comment by ColdGrub1384** [comment]
Here is a SciPy build for ARM64: https://github.com/goodclass/scipy_arm64. Some things don't work and make the app crash. After SciPy, SciKit-Learn will be easy to build and include.

**Comment by holzschu** [comment]
Thanks. Of course, compiling is only half the work. Making sure allocated memory is released when the kernel is terminated is the hard part for Carnets.

**Comment by ifuchs** [comment]
Just checking as to whether there is any progress toward adding Scikit-Learn.

**Comment by ifuchs** [comment]
New beta has scipy but scikit-learn does not appear to be installable.

**Comment by holzschu** [comment]
Yes, scikit-learn is an even bigger package than scipy. It requires scipy, but it also has its own C/Cython modules to cross-compile and check for memory issues. But we're getting there, slowly but surely. 

Keep in mind that iOS terminates apps that use more than 100% CPU for more than 180s, so it might not be the best platform for intensive machine learning.

**Comment by holzschu** [comment]
The version with scikit-learn is now available for TestFlight: https://testflight.apple.com/join/UbNIjkjp

**Comment by holzschu** [comment]
The version with scikit-learn is now available on the AppStore: https://apps.apple.com/fr/app/carnets-jupyter-with-scipy/id1559497253
I am going to be able to close this issue.
